File: challenges/ruby/block-member.rb
load 'library-data.rb'

def block_member(library_data, email)
  information_path = ['userManagement', 'members', email, 'isBlocked']
  library_data.set_in(information_path, true)
end

updated_library_data = block_member($library_data, 'samantha@gmail.com')
information_path = ['userManagement', 'members', 'samantha@gmail.com', 'isBlocked']
pp [updated_library_data.dig(*information_path), $library_data.dig(*information_path)]

File: challenges/elixir/lib/block_member.ex
defmodule BlockMember do
  @doc """
  A function that receives library data and an email address and returns a new version
  of library data without altering the original version, where the user with the given email is blocked.

  ## Examples

      iex> updated_library_data = BlockMember.block_member(LibraryData.library_data(), "samantha@gmail.com")
      iex> information_path = ["userManagement", "members", "samantha@gmail.com", "isBlocked"]
      iex> [get_in(updated_library_data, information_path), get_in(LibraryData.library_data(), information_path)]
      [true, false]
  """
  def block_member(library_data, email) do
    information_path = ["userManagement", "members", email, "isBlocked"]
    put_in(library_data, information_path, true)
  end
end
